.github/workflows/ubuntu-ci.yml

```diff
@@ -0,0 +1,51 @@
+name: Ubuntu-CI
+on: [push, pull_request]
+
+jobs:
+  ubuntu-build:
+    if:
+      contains(github.repository, 'SFTtech')==false 
+    strategy:
+      matrix:
+        cc: [gcc, clang]
+    runs-on: ubuntu-latest
+    env:
+      CC: ${{matrix.cc}}
+    steps:
+      - name: Checkout sources
+        uses: actions/checkout@v1
+      - name: Setup apt repositories
+        run: |
+          if [ $CC = clang ]; then
+            sudo add-apt-repository "deb http://apt.llvm.org/bionic/ llvm-toolchain-bionic-9 main"
+          fi
+      - name: Update apt repositories
+        run: sudo apt-get update
+      - name: Install Dependencies
+        run: |
+          sudo apt-get install -y --no-install-recommends ccache cmake cython3 libeigen3-dev libepoxy-dev libfontconfig1-dev libfreetype6-dev libharfbuzz-dev libogg-dev libopus-dev libopusfile-dev libpng-dev libsdl2-dev libsdl2-image-dev python3-dev python3-jinja2 python3-numpy python3-pil python3-pip python3-pygments pylint3 python3-pycodestyle qml-module-qtquick-controls qtdeclarative5-dev flex make
+      - name: Install C compiler
+        run: |
+          if [ $CC = clang ]; then
+            sudo apt-get install -y --no-install-recommends clang-9
+            export CXX=clang++
+          else
+            sudo apt-get install -y --no-install-recommends $CC g++
+          fi
+      - name: Sanity check
+        run: make checkall # This is skipped as long as there are issues in the main repo.
+        if: 
+          contains(github.event, 'pull_request')
+      - name: Generate build files
+        run: |
+          if [ $CC = clang ]; then
+            ./configure --compiler=clang++ --mode=debug --flags="-Werror" --ccache --download-nyan
+          else
+            ./configure --mode=debug --flags="-Werror" --ccache --download-nyan 
```

Comment:
I've added Ccache to the Ubuntu and MacOSX builds, for future use within PRs, where we could cache the `build-dir` after the first build and then pull in the newer commits and run configure again, speeding up the whole process. Also for people in their forks something like this might be a nice feature.